platform/bootstrap/bootstrap.yml

```diff
@@ -300,36 +453,69 @@ Resources:
       Path: /
       Roles:
       - Ref: ClusterRole
-  BootstrapInstance:
-    Type: AWS::EC2::Instance
-    DependsOn: PublicSubnet1
+  AutoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    DependsOn:
+      - PublicSubnet1
+      - PublicSubnet2
+      - PublicSubnet3
+    UpdatePolicy:
+      AutoScalingRollingUpdate:
+        MaxBatchSize: 1
+        MinInstancesInService: !Ref ManagerSize
+        PauseTime: PT20M
+        WaitOnResourceSignals: true
     Properties:
-      SubnetId: !Ref PublicSubnet1
-      SecurityGroupIds:
-        - !GetAtt SecurityGroup.GroupId
-      AvailabilityZone: !GetAtt PublicSubnet1.AvailabilityZone
-      IamInstanceProfile: !Ref ProvisionerInstanceProfile
-      ImageId:
-        Fn::FindInMap:
-        - AMI
-        - Ref: AWS::Region
-        - Ubuntu
-      InstanceType: !Ref BootstrapInstanceType
-      PrivateIpAddress: "192.168.2.254"
-      KeyName: !Ref BootstrapKeyName
+      DesiredCapacity: !Ref ManagerSize
+      HealthCheckGracePeriod: 200
+      HealthCheckType: EC2
+      LaunchConfigurationName: !Ref AsgLaunchConfig
+      MaxSize: 5
+      MinSize: 0
       Tags:
       - Key: Name
+        PropagateAtLaunch: true
         Value:
           Fn::Join:
           - '-'
           - - Ref: AWS::StackName
-            - controller
-      - Key: infrakit.group
-        Value: swarm-managers
-      - Key: infrakit.role
-        Value: managers
+            - manager
+      - Key: Deployment
+        PropagateAtLaunch: true
+        Value: Infrakit
+      - Key: Role
+        PropagateAtLaunch: true
+        Value: manager
+      VPCZoneIdentifier:
+      - Fn::Join:
+        - ','
+        -  - !Ref PublicSubnet1
+           - !Ref PublicSubnet2
+           - !Ref PublicSubnet3
+  AsgLaunchConfig:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    DependsOn:
+      - ProvisionerInstanceProfile
+      - SecurityGroup
+    Properties:
+      AssociatePublicIpAddress: true
+      BlockDeviceMappings:
+        - DeviceName: /dev/xvdb
```

Comment:
I'm confused. you can use ephemeral disk (vs EBS), which is what the /dev/xvd[a-e] range is reserved for, and you can't set the size and type for ephemerals. But be aware ephemeral disks are not supported anymore on recent instance types.